api/python/provisioner/commands/check.py

```diff
@@ -1104,6 +1053,60 @@ def _controller_ip(*, args: str) -> Union[CheckEntry, List[CheckEntry]]:
 
         return res
 
+    def _validate_bmc_check(self, check: str):
+
+        if check == self.STONITH:
+            bmc_check = cfg.Checks.BMC_STONITH.value
+        elif check == self.ACCESSIBLE:
+            bmc_check = cfg.Checks.BMC_ACCESSIBILITY.value
+        else:
+            raise ValueError(f'BMC Check "{check}" is not supported')
+
+        check_entry: CheckEntry = CheckEntry(bmc_check)
+        # Check for import errors
+        if cortx_py_utils_import_error:
+            check_entry.set_fail(
+                checked_target=cfg.ALL_MINIONS,
+                comment="Package cortx-py-utils not installed")
+            return check_entry
+
+        try:
+            # Get node list
+            nodes = Check._get_pillar_data("cluster/node_list")
+        except Exception as exc:
+            check_entry.set_fail(checked_target=cfg.ALL_MINIONS,
+                                 comment=str(exc))
+            return check_entry
+
+        res: List[CheckEntry] = list()
+        _DECRYPT_PSWD_CMD = ('salt-call lyveutil.decrypt cluster {val}'
+                             ' --output=newline_values_only')
+
+        for node in nodes:
+            check_res: CheckEntry = CheckEntry(bmc_check)
+
+            try:
+                # Get BMC IP, user and secret.
+                bmc_ip = Check._get_pillar_data(f"cluster/{node}/bmc/ip")
+                bmc_user = Check._get_pillar_data(f"cluster/{node}/bmc/user")
+                secret = Check._get_pillar_data(f"cluster/{node}/bmc/secret")
+
+                # Decrypt the secret
+                cmd = _DECRYPT_PSWD_CMD.format(val=secret)
+                minion_id = local_minion_id()
```

Comment:
makes sense to move outside the loop since it should be always the same

Reply:
comment is addressed